MonoGame.Framework/Quaternion.cs

```diff
@@ -342,45 +522,76 @@ public static void Divide(ref Quaternion quaternion1, ref Quaternion quaternion2
 		    result.Y = ((y * num) + (num3 * w)) + num12;
 		    result.Z = ((z * num) + (num2 * w)) + num11;
 		    result.W = (w * num) - num10;
-
         }
 
+        #endregion
+
+        #region Dot
 
+        /// <summary>
+        /// Returns a dot product of two quaternions.
+        /// </summary>
+        /// <param name="quaternion1">The first quaternion.</param>
+        /// <param name="quaternion2">The second quaternion.</param>
+        /// <returns>The dot product of two quaternions.</returns>
         public static float Dot(Quaternion quaternion1, Quaternion quaternion2)
         {
             return ((((quaternion1.X * quaternion2.X) + (quaternion1.Y * quaternion2.Y)) + (quaternion1.Z * quaternion2.Z)) + (quaternion1.W * quaternion2.W));
         }
 
-
+        /// <summary>
+        /// Returns a dot product of two quaternions.
+        /// </summary>
+        /// <param name="quaternion1">The first quaternion.</param>
+        /// <param name="quaternion2">The second quaternion.</param>
+        /// <param name="result">The dot product of two quaternions as an output parameter.</param>
         public static void Dot(ref Quaternion quaternion1, ref Quaternion quaternion2, out float result)
         {
             result = (((quaternion1.X * quaternion2.X) + (quaternion1.Y * quaternion2.Y)) + (quaternion1.Z * quaternion2.Z)) + (quaternion1.W * quaternion2.W);
         }
 
+        #endregion
+
+        #region Equals
 
+        /// <summary>
+        /// Compares whether current instance is equal to specified <see cref="Object"/>.
+        /// </summary>
+        /// <param name="obj">The <see cref="Object"/> to compare.</param>
+        /// <returns><c>true</c> if the instances are equal; <c>false</c> otherwise.</returns>
         public override bool Equals(object obj)
         {
-             bool flag = false;
-		    if (obj is Quaternion)
-		    {
-		        flag = this.Equals((Quaternion) obj);
-		    }
-		    return flag;
+            return (obj is Quaternion) ? this == (Quaternion)obj : false;
```

Comment:
If you are going to change this we might as well make the code clear and simple.
```
if (obj is Quaternion)
   return Equals((Quaternion)obj);
return false;
```